### PR TITLE
Problem: sdk v0.46.11 is not used

### PR DESCRIPTION
**Pull request by yihuang**
Solution:
- update sdk/tendermint/ethermint dependencies
- patch cometbft-db to use grocksdb
- still need to replace tm-db because of indirect dependencies
- prepare v1.0.5 release in this PR.

👮🏻👮🏻👮🏻 !!!! REFERENCE THE PROBLEM YOUR ARE SOLVING IN THE PR TITLE AND DESCRIBE YOUR SOLUTION HERE !!!! DO NOT FORGET !!!! 👮🏻👮🏻👮🏻

# PR Checklist:

- [ ] Have you read the [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md)?
- [ ] Does your PR follow the [C4 patch requirements](https://rfc.zeromq.org/spec:42/C4/#23-patch-requirements)?
- [ ] Have you rebased your work on top of the latest master? 
- [ ] Have you checked your code compiles? (`make`)
- [ ] Have you included tests for any non-trivial functionality?
- [ ] Have you checked your code passes the unit tests? (`make test`)
- [ ] Have you checked your code formatting is correct? (`go fmt`)
- [ ] Have you checked your basic code style is fine? (`golangci-lint run`)
- [ ] If you added any dependencies, have you checked they do not contain any known vulnerabilities? (`go list -json -m all | nancy sleuth`)
- [ ] If your changes affect the client infrastructure, have you run the integration test?
- [ ] If your changes affect public APIs, does your PR follow the [C4 evolution of public contracts](https://rfc.zeromq.org/spec:42/C4/#26-evolution-of-public-contracts)?
- [ ] If your code changes public APIs, have you incremented the crate version numbers and documented your changes in the [CHANGELOG.md](https://github.com/crypto-org-chain/chain-main/blob/master/CHANGELOG.md)?
- [ ] If you are contributing for the first time, please read the agreement in [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md) now and add a comment to this pull request stating that your PR is in accordance with the [Developer's Certificate of Origin](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md#developer-certificate-of-originn).

Thank you for your code, it's appreciated! :)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -10,6 +10,7 @@
 
 - [#904](https://github.com/crypto-org-chain/cronos/pull/904) Enable "dynamic-level-bytes" on new `application.db`.
 - [#907](https://github.com/crypto-org-chain/cronos/pull/907) Apply a configurable limit in rpc apis.
+- [#]() Update to cosmos-sdk v0.46.11.
```
Comment (yihuang) [inline CHANGELOG.md:13]:
```suggestion
- [#909](https://github.com/crypto-org-chain/cronos/pull/909) Update to cosmos-sdk v0.46.11.
```

File: go.mod [hunk 2]
```diff
@@ -193,12 +194,14 @@ require (
 )
 
 replace (
+	// https://github.com/cometbft/cometbft-db/pull/42
+	github.com/cometbft/cometbft-db => github.com/yihuang/cometbft-db v0.7.1-0.20230306024652-3af958bced3e
```
Comment (yihuang) [inline go.mod:198]: https://github.com/cometbft/cometbft-db/pull/42

Comment (tomtau) [reply go.mod:198]: forked here: https://github.com/crypto-org-chain/cometbft-db if expected cronos PR needs to be merged before the upstream cometbft-db patch

Comment (yihuang) [reply go.mod:198]: let's use fork first

Comment (yihuang) [reply go.mod]: Just added v1.0.5 prepare directly in this PR.

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,6 +1,8 @@
 # Changelog
 
-## UNRELEASED
+*Mar 6, 2023
```
Comment (mmsqe) [inline CHANGELOG.md:3]:
```suggestion
*Mar 6, 2023*
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -10,8 +12,9 @@
 
 - [#904](https://github.com/crypto-org-chain/cronos/pull/904) Enable "dynamic-level-bytes" on new `application.db`.
 - [#907](https://github.com/crypto-org-chain/cronos/pull/907) Apply a configurable limit in rpc apis.
+- [#909](https://github.com/crypto-org-chain/cronos/pull/909) Update to cosmos-sdk v0.46.11.
 
-*Feb 15, 2022*
+*Feb 15, 2023
```
Comment (mmsqe) [inline CHANGELOG.md:17]:
```suggestion
*Feb 15, 2023*
```

**Review by mmsqe — APPROVED** [review]
only date format need update

**Comment by yihuang** [comment]
> only date format need update

yeah, fixed, thanks.

**Review by calvinaco — COMMENTED** [review]
Checking Cosmos SDK v0.46.11 CHANGELOG
https://github.com/cosmos/cosmos-sdk/blob/v0.46.11/CHANGELOG.md

Want to check if this store change is potentially breaking or not?
https://github.com/cosmos/cosmos-sdk/pull/14410

**Comment by yihuang** [comment]
> Checking Cosmos SDK v0.46.11 CHANGELOG https://github.com/cosmos/cosmos-sdk/blob/v0.46.11/CHANGELOG.md
> 
> Want to check if this store change is potentially breaking or not? [cosmos/cosmos-sdk#14410](https://github.com/cosmos/cosmos-sdk/pull/14410)

should be non-breaking, we had the uneven case in some nodes before, which eventually manifested as app-hash mismatch, but not in the chain itself.

File: go.mod [hunk 5]
```diff
@@ -193,12 +194,14 @@ require (
...
+	// https://github.com/cometbft/cometbft-db/pull/42
+	github.com/cometbft/cometbft-db => github.com/crypto-org-chain/cometbft-db v0.0.0-20230306031617-b3e4fd3331c4
 	// Ref: https://forum.cosmos.network/t/ibc-security-advisory-dragonberry/7702
 	github.com/confio/ics23/go => github.com/confio/ics23/go v0.9.0
-	github.com/cosmos/cosmos-sdk => github.com/cosmos/cosmos-sdk v0.46.10
+	github.com/cosmos/cosmos-sdk => github.com/cosmos/cosmos-sdk v0.46.11
 	github.com/crypto-org-chain/cronos/versiondb => ./versiondb
 	github.com/ethereum/go-ethereum => github.com/crypto-org-chain/go-ethereum v1.10.19-deepcopy-jumptable
-	github.com/evmos/ethermint => github.com/crypto-org-chain/ethermint v0.20.7-cronos
+	github.com/evmos/ethermint => github.com/crypto-org-chain/ethermint v0.20.8-cronos
```
Comment (yihuang) [inline go.mod:204]: https://github.com/crypto-org-chain/ethermint/compare/v0.20.7-cronos...v0.20.8-cronos

Comment (mmsqe) [reply go.mod:204]: I get a diff on gomod2nix.toml for current v0.20.8-cronos

**Review by tomtau — APPROVED** [review]

File: go.mod (hunk 5 above)
Comment (yihuang) [reply go.mod:204]: I see, but it should be ok, since it's not used here ;D

**Pull request merged**